lib/buildpack/compile/scripts_parser.rb

```diff
@@ -64,6 +70,40 @@ def scripts_section_exists?(check_commands)
       return_value
     end
 
+    def xml_scripts_section_exists?(check_commands)
+      check_keys = %w(PrecompileScript PostcompileScript)
```

Comment:
MSBuild further complicates this feature, the names PrecompileScript and PostcompileScript have no special meaning, it is actually the `BeforeTargets` and `AfterTargets` attributes which also appear with them that tell them what to do.  These attributes are semi-colon delimited lists of commands. For example:
`<Target Name="ThisNameDoesNotMatter" BeforeTargets="Compile;Publish">`
Will run these commands before any compile or publish command.

There is a tag name which will run these commands before only compile as well, that tag is `BeforeCompile`.  For example:
`<Target Name="BeforeCompile">`
Will run commands before the compile step.

Examples of post-compile actions in each format:
`<Target Name="ThisNameDoesNotMatter" AfterTargets="Compile;Publish">`
`<Target Name="AfterCompile">`